src/Laravel/Eloquent/Metadata/Factory/Property/EloquentPropertyMetadataFactory.php

```diff
@@ -93,7 +97,7 @@ public function create(string $resourceClass, string $property, array $options =
             }
 
             $collection = false;
-            if (HasMany::class === $relation['type']) {
+            if (\in_array($relation['type'], [HasMany::class, HasManyThrough::class, BelongsToMany::class, MorphMany::class, MorphToMany::class], true)) {
```

Comment:
Nit: using an associative array with types as keys and `true` as values, or just testing all conditions directly in the if (separated by `||`) will be a bit faster than using `in_array()`.